typing/parmatch.ml

```diff
@@ -1008,35 +1139,45 @@ let rec satisfiable pss qs = match pss with
           satisfiable pss (q::qs)
     | {pat_desc = (Tpat_any | Tpat_var(_))}::qs ->
         let pss = simplify_first_col pss in
-        let q0 = discr_pat omega pss in
-        begin match build_specialized_submatrices ~extend_row:(@) q0 pss with
-        | { default; constrs = [] } ->
-          (* first column of pss is made of variables only *)
-          satisfiable default qs
-        | { default; constrs }  ->
-            if full_match false constrs then
-              List.exists
-                (fun (p,pss) ->
-                  not (is_absent_pat p) &&
-                  satisfiable pss (simple_match_args p omega @ qs))
-                constrs
-            else
+        if not (first_column_is_coherent pss) then (
```

Comment:
Here `q` is `omega`, so `coherent pss` is the same as `coherent (q :: pss)`, and this test is correct -- but a comment could point that out.

Reply:
I'm starting to feel like we have enough comments that really simple ones (like this one would be) can actually be omitted (I believe that, as is, one can fairly quickly recover the reason for not checking `q :: pss`)

Reply:
Your code, your choice.